Source/WebCore/Modules/speech/SpeechRecognition.cpp

```diff
@@ -55,9 +55,9 @@ Ref<SpeechRecognition> SpeechRecognition::create(Document& document)
 SpeechRecognition::SpeechRecognition(Document& document)
     : ActiveDOMObject(document)
 {
-    if (auto* page = document.page()) {
+    if (RefPtr page = document.page()) {
         m_connection = page->speechRecognitionConnection();
-        m_connection->registerClient(*this);
+        protectedConnection()->registerClient(*this);
```

Comment:
m_connection is only ever assigned once, and the class holds a Ref on it. So I think we should be able to remove protectedConnection(), make m_connection const and initialize it with lazyInitialize().

Reply:
Nice catch. I'll look for this a bit more carefully going forward.